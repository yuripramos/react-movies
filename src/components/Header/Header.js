import React, { Component } from "react";
import { shape, func } from "prop-types";
import { Container, Row, Column } from "../../styles/grid";
import SidebarMenu from "../common/SidebarMenu";
import Logo from "../../assets/images/logo.png";
import { HeaderWrapper, TitleWrapper } from "./styles";
import { slide as Menu } from "react-burger-menu";
import Hide from "../common/Hide";
import Icon from "../common/Icon";
import { rem } from "../../styles/tools";
import { white, black } from "../../styles/settings";

const bmStylesClosed = {
  bmBurgerButton: {
    display: "inline-block",
    position: "absolute",
    top: `${rem(20)}`,
    left: `${rem(25)}`,
    padding: `${rem(2)}`,
    cursor: "pointer"
  },
  bmMenuWrap: {
    left: 0
  }
};

const bmStylesOpened = {
  bmBurgerButton: {
    display: "inline-block",
    position: "absolute",
    top: `${rem(20)}`,
    right: `${rem(15)}`,
    padding: `${rem(2)}`,
    cursor: "pointer"
  },
  bmMenuWrap: {
    left: 0
  },
  bmOverlay: {
    left: 0,
    top: 0
  }
};

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuOpen: false
    };

    this.onToggleMenu = this.onToggleMenu.bind(this);
    this.closeMenu = this.closeMenu.bind(this);
  }
  onToggleMenu({ isOpen }) {
    this.setState({ menuOpen: isOpen });
  }
  closeMenu() {
    this.setState({ menuOpen: false });
  }

  render() {
    const { history } = this.props;
    const { menuOpen } = this.state;

    return (
      <HeaderWrapper>
        <Container onClick={this.closeMenu}>
          <Hide below="lg">
            <Row>
              <Column>
                <SidebarMenu
                  currentLocation={history.location.pathname}
                  desktop
                />
              </Column>
              <Column>
                <TitleWrapper>
                  <img
                    src={Logo}
                    alt="business logo"
                    width="44px"
                    height="44px"
                  />
                </TitleWrapper>
              </Column>
            </Row>
          </Hide>
          <Hide above="lg">
            <Menu
              isOpen={menuOpen}
              onStateChange={this.onToggleMenu}
              customBurgerIcon={
                menuOpen === true ? (
                  <Icon name="MenuClose" color={white} />
                ) : (
                  <Icon name="Menu" color={black} />
                )
              }
              width="90%"
              styles={menuOpen === true ? bmStylesOpened : bmStylesClosed}
            >
              <SidebarMenu
                closeMenu={this.closeMenu}
                currentLocation={history.location.pathname}
              />
            </Menu>
            <TitleWrapper>
              <img src={Logo} alt="business logo" width="50px" height="50px" />
            </TitleWrapper>
          </Hide>
        </Container>
      </HeaderWrapper>
    );
  }
}

export default Header;

Header.propTypes = {
  history: shape({
    push: func
  })
};
