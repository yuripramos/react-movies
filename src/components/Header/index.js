import Header from "./Header";
import { connect } from "redux-zero/react";
import { withRouter } from "react-router-dom";

import articlesActions from "../../actions/Articles";

export default connect(
  ({ getAuthors, getArticles, authorsList }) => ({
    getAuthors,
    getArticles,
    authorsList
  }),
  articlesActions
)(withRouter(Header));
