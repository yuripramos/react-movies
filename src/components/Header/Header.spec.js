import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

let props;

beforeEach(() => {
  props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: ""
      }
    }
  };
});

describe("Header component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<Header {...props} />)).toMatchSnapshot();
  });

  describe("onToggleMenu", () => {
    it("should update the state", () => {
      const instance = shallow(<Header {...props} />).instance();

      instance.onToggleMenu({ isOpen: true });

      expect(instance.state.menuOpen).toBe(true);
    });
  });

  describe("closeMenu", () => {
    it("should update the state", () => {
      const instance = shallow(<Header {...props} />).instance();

      instance.closeMenu();

      expect(instance.state.menuOpen).toBe(false);
    });
  });
});
