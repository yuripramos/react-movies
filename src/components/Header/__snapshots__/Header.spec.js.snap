// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot with props 1`] = `
<styled.header>
  <Container
    blacklist={
      Array [
        "css",
        "width",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "fontSize",
        "color",
        "bg",
        "flex",
        "order",
        "alignSelf",
      ]
    }
    onClick={[Function]}
  >
    <Hide
      above={null}
      below="lg"
    >
      <Row>
        <Column>
          <SidebarMenu
            currentLocation=""
            desktop={true}
          />
        </Column>
        <Column>
          <styled.div>
            <img
              alt="business logo"
              height="44px"
              src="test-file-stub"
              width="44px"
            />
          </styled.div>
        </Column>
      </Row>
    </Hide>
    <Hide
      above="lg"
      below={null}
    >
      <Menu
        bodyClassName=""
        burgerBarClassName=""
        burgerButtonClassName=""
        className=""
        crossButtonClassName=""
        crossClassName=""
        customBurgerIcon={
          <IconGenerator
            color="#333"
            name="Menu"
          />
        }
        disableAutoFocus={false}
        disableCloseOnEsc={false}
        htmlClassName=""
        id=""
        isOpen={false}
        itemClassName=""
        itemListClassName=""
        menuClassName=""
        morphShapeClassName=""
        noOverlay={false}
        onStateChange={[Function]}
        outerContainerId=""
        overlayClassName=""
        pageWrapId=""
        styles={
          Object {
            "bmBurgerButton": Object {
              "cursor": "pointer",
              "display": "inline-block",
              "left": "1.5625rem",
              "padding": "0.125rem",
              "position": "absolute",
              "top": "1.25rem",
            },
            "bmMenuWrap": Object {
              "left": 0,
            },
          }
        }
        width="90%"
      >
        <SidebarMenu
          closeMenu={[Function]}
          currentLocation=""
        />
      </Menu>
      <styled.div>
        <img
          alt="business logo"
          height="50px"
          src="test-file-stub"
          width="50px"
        />
      </styled.div>
    </Hide>
  </Container>
</styled.header>
`;
